SOURCES/etc/xapi.d/plugins/updater.py

```diff
@@ -1,16 +1,84 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
+from functools import wraps
+import ConfigParser
+import json
+import os
 import subprocess
 import sys
 import traceback
 import XenAPIPlugin
 import yum
-import ConfigParser
-import json
 
 sys.path.append('.')
-from xcpngutils import configure_logging
+from xcpngutils import configure_logging, run_command
+from xcpngutils.filelocker import FileLocker
+
+
+REPOS = ('xcp-ng-base', 'xcp-ng-updates')
+
+
+class OperationException(Exception):
+    pass
+
+class OperationLocker(FileLocker):
+    __slots__ = ('operation', 'current_operation')
+
+    def __init__(self, operation, timeout=0):
+        super(OperationLocker, self).__init__(
+            timeout=timeout,
+            auto_remove=False,
+            dir=os.path.dirname(os.path.abspath(__file__))
```

Comment:
Does this mean that a lock file will be created in `/etc/xapi.d/plugins/updater.py`? If so, this does not seem the best place (`/etc`).

Reply:
So where is the best place?
In my filesystem there are locks in `/etc/udev`, `/home/<user>/.local`, `/run`, `/usr/lib`, `/var/lib`.